src/main/java/io/quarkus/bot/PingWhenNeedsTriageRemoved.java

```diff
@@ -0,0 +1,80 @@
+package io.quarkus.bot;
+
+import io.quarkiverse.githubapp.ConfigFile;
+import io.quarkiverse.githubapp.event.Issue;
+import io.quarkus.bot.config.QuarkusBotConfig;
+import io.quarkus.bot.config.QuarkusBotConfigFile;
+import io.quarkus.bot.util.Labels;
+import io.quarkus.bot.util.Patterns;
+import io.quarkus.bot.util.Strings;
+import org.jboss.logging.Logger;
+import org.kohsuke.github.GHEventPayload;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHLabel;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class PingWhenNeedsTriageRemoved {
+    private static final Logger LOG = Logger.getLogger(PingWhenNeedsTriageRemoved.class);
+
+    @Inject
+    QuarkusBotConfig quarkusBotConfig;
+
+    void pingWhenNeedsTriageRemoved(@Issue.Unlabeled GHEventPayload.Issue issuePayload,
+            @ConfigFile("quarkus-bot.yml") QuarkusBotConfigFile quarkusBotConfigFile) throws IOException {
+
+        if (quarkusBotConfigFile == null) {
+            LOG.error("Unable to find triage configuration.");
+            return;
+        }
+
+        GHIssue issue = issuePayload.getIssue();
+        GHLabel removedLabel = issuePayload.getLabel();
+
+        if (!removedLabel.toString().equals(Labels.TRIAGE_NEEDS_TRIAGE)) {
+            return;
+        }
+
+        Set<String> mentions = new TreeSet<>();
+
+        for (QuarkusBotConfigFile.TriageRule rule : quarkusBotConfigFile.triage.rules) {
+            if (matchRule(issue, rule)) {
+                if (!rule.notify.isEmpty()) {
+                    for (String mention : rule.notify) {
+                        if (!mention.equals(issue.getUser().getLogin())) {
+                            mentions.add(mention);
+                        }
+                    }
+                }
+            }
+        }
+
+        if (removedLabel.getName().equals(Labels.TRIAGE_NEEDS_TRIAGE)) {
```

Comment:
Since have already checked in the beginning that if the label is not `needs-triage` then simply return, so here this check can be skipped :)

Reply:
That's the whole point of discarding things early as I explained it: you can drop this and simplify your main logic.